### PR TITLE
Add audio

### PR DESCRIPTION
**Pull request by AdrienLemaire**
This branch is a replacement for the current **audio** branch, which is too outdated since 2013 Dec.

### CODE REVIEW

**Comment by AdrienLemaire** [comment]
Question about the translations. Should I create a related branch in habitrpg-shared, and add all the new strings (without their translation) in the locale json files?

**Comment by AdrienLemaire** [comment]
@lefnire @paglias How do I add new translatable strings in the code ?

I added `env.t("music")` in the jade file, then tried to add the translation in each of those files:
- public/bower_components/habitrpg-shared/locales/en/generic.json
- public/bower_components/habitrpg-shared/locales/en_GB/generic.json
- node_modules/habitrpg-shared/locales/en/generic.json
- node_modules/habitrpg-shared/locales/en_GB/generic.json

But it cannot find the string:

error

The wikia doc is lacking for this kind of information :)

**Comment by Alys** [comment]
@Fandekasp : To edit translation strings, edit the files under only node_modules/habitrpg-shared/locales/en/ and then run grunt while in node_modules/habitrpg-shared. Don't edit other translation languages (not even en_GB) because that's all handled in Transifex. Don't try editing bower_components manually because they're all generated, if I understand correctly. I've had success with editing just the locales/en files and then running grunt, and also restarting npm when necessary.

**Comment by AdrienLemaire** [comment]
Thanks @Alys . So I'll have to create a related branch in habitrpg-shared for the translations. Got it!

**Comment by AdrienLemaire** [comment]
[Wikia doc updated](http://habitrpg.wikia.com/wiki/Guidance_for_Blacksmiths#Translatable_strings)

**Comment by AdrienLemaire** [comment]
New PR [#275](https://github.com/HabitRPG/habitrpg-shared/pull/275) created in **habitrpg-shared**

**Comment by AdrienLemaire** [comment]
@lefnire I believe this branch and its counter-part on habitrpg-shared can be merged after you review it. You can also remove the useless branch **habitrpg/audio**.
You'll have to run the migration script `migrations/20140809_soundTheme.js` after merging the code.

There are a lot of things that can be added as audio features, but they can be added later, in new branches. Please do not wait another 6 months to merge that branch, thanks ^^

**Comment by Alys** [comment]
@Fandekasp : Does that migration script enable sound for all users? If not, ignore this comment. :) But if it does turn on sound automatically for everyone, I suggest that it should be changed so that users must manually opt-in. There'll be some users who find website sound effects annoying and will want to have full control over them from the beginning. There'll be people who are using Habit at work where sudden unexpected sounds will be unprofessional, or at school where it will be distracting to other students.

**Comment by AdrienLemaire** [comment]
Yes, the migration script enable sound for all users. I don't know how to add a specific window to ask users to update that setting the first time, nor do I have the time to search how to do unfortunately.

I'll let @lefnire handle it (most simple solution: disable sound for all users in migration script, and let people know of the feature in a bailey news)

**Comment by Alys** [comment]
Notifying users by a Bailey announcement and letting them choose whether to enable sound is by far the preferred option.

Can you please either edit the migration script to disable sound by default or remove the migration script so that it won't be merged? I think it's likely that the migration script is actually not needed so removal is probably best. If I understand the code correctly, your changes to src/models/user.js will be applied to all existing users automatically, so that alone would do the "migration".

**Comment by AdrienLemaire** [comment]
done

**Comment by SabreCat** [comment]
Tagging @DanielTheBard so he's aware this is underway!

**Comment by DanielTheBard** [comment]
Awesome :)

**Comment by lefnire** [comment]
Will hit this when back from vacay

File: src/models/user.js [hunk 1]
```diff
@@ -255,6 +255,8 @@ var UserSchema = new Schema({
     skin: {type:String, 'default':'915533'},
     shirt: {type: String, 'default': 'blue'},
     timezoneOffset: Number,
+    sound: Boolean,
+    soundTheme: {type:String, 'default':'danielTheBard'},
```
Comment (lefnire) [inline src/models/user.js:259]: I may consolidate this to simply

``` javascript
sound: {type:String, 'default': 'danielTheBard', enum: ['off','danielTheBard']},
```

going down the list, will comment as I go

**Comment by lefnire** [comment]
Great, merging! I'm gonna rework it a bit and ping you back with my changes. Cool to delete any `audio` branches then?

**Comment by AdrienLemaire** [comment]
Yes, you can delete `audio`

**Comment by lefnire** [comment]
Ok, pushed to beta. If you get the "Validation Error" thing, just try back at a later time. Forgot we had some vestigial `preferences.sound: Boolean` saved from our previous beta attempt, running migration

**Comment by DanielTheBard** [comment]
How do we feel about the volume levels?

**Comment by AdrienLemaire** [comment]
Maybe a bit loud. But I think we should add later on a feature to user-adjust the volume, so it's not a big deal for now.

**Comment by DanielTheBard** [comment]
@Fandekasp - Yeah, I tried to use other sites with sound as a reference. Usually, my computer at half volume was a comfortable level, and max volume was a tad loud.

**Comment by DanielTheBard** [comment]
Not hearing the chat sound when I send a chat message.

**Comment by AdrienLemaire** [comment]
Not implemented, since it's on a different page, sorry. Will take a look
later when I get some tiime

On 16 August 2014 10:25, Daniel the Bard notifications@github.com wrote:

> Not hearing the chat sound when I send a chat message.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/HabitRPG/habitrpg/pull/3856#issuecomment-52378192.

## 

Cheers,
Adrien LEMAIRE
_Director of Technology - Aquasys http://www.aquasys.co.jp_
_Senior Engineer - Ikkyo Technology http://ikkyotech.com_
_Google+ http://gplus.to/Fandekasp  CV
http://vizualize.me/adrien.lemaire_

**Pull request merged**